src/PieChart/PieArcSeries/PieArcSeries.tsx

```diff
@@ -124,36 +179,44 @@ export class PieArcSeries extends Component<PieArcSeriesProps> {
   }
 
   centroid(innerRadius: number, outerRadius: number) {
-    return (data) => {
+    return (data: ArcData) => {
       const newOuter = this.calculateOuterRadius(outerRadius, data);
 
-      return arc()
+      return arc<any, ArcData>()
         .innerRadius(innerRadius)
         .outerRadius(newOuter)
         .centroid(data);
     };
   }
 
-  outerArc(outerRadius: number) {
-    return arc()
+  outerArc(outerRadius: number): Arc<any, ArcData> {
+    return arc<any, ArcData>()
       .innerRadius(outerRadius * factor)
       .outerRadius(outerRadius * factor);
   }
 
   render() {
-    const { animated, label, arc, data } = this.props;
+    const { animated, label, arc, data, width } = this.props;
 
     const { outerRadius, innerRadius } = this.calculateRadius();
     const innerArc = this.innerArc(innerRadius, outerRadius);
     const outerArc = this.outerArc(outerRadius);
-    const positions = this.calculateLabelPositions(outerArc, outerRadius);
+    // TODO: is there any way to get this from `outerArc` object?
+    const outerArcRadius = outerRadius * factor;
+    const positions = label
+      ? this.calculateLabelPositions({
+          outerArc,
+          outerArcRadius,
+          minDistance: label.props.height + 4
```

Comment:
Where does this `4` come from?

Reply:
This a vertical margin between labels. Previous `minDistance` was 15 (11 + 4). I'm not sure whether to leave this 4 pixels hardcoded or add it as new label prop - `margin`.

There is also an alternative way: remove that margin completely and require the user to include all the margins in `height` prop

Reply:
I'd prefer user not having to pass these things as much as possible.